src/body/body_reader.rs

```diff
@@ -26,7 +23,7 @@ pub struct BodyData {
 
 #[must_use = "futures do nothing unless you `.await` or poll them"]
 pub struct BodyReader {
```

Comment:
I prefer the `http_body_util::Collected` struct to `via::body::BodyData` but it cannot be constructed from it's arguments and I prefer `via::body::BodyReader` to `http_body_util::combinators::Collect`.